src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -827,6 +854,47 @@ abstract class ManagedStrategy(
         return (threadDescriptor.eventTracker === this)
     }
 
+    /**
+     * Aborts all threads in case if all `TestThread`s are in state `FINISHED`
+     * and all running user threads are `LiveLocked` or `Parked`.
+     *
+     * This method is expected to be called before [onSwitchPoint] method in
+     * order to abort execution and not to create unnecessary switch points.
+     *
+     * Since this method only can abort executions with `LiveLocked` or `Parked`
+     * user-threads and `Finished` test threads, then invocations of this method are
+     * only meaningful near the corresponding strategy hooks
+     * ([ManagedStrategy.processLoopDetectorDecision], [ManagedStrategy.park],
+     * and [ManagedStrategy.onThreadFinish] where otentially test thread could finish).
+     *
+     * @param threadId id of thread that invoked this method.
+     * @param blockingReason blocking reason of invoking thread (determined by strategy) if exists.
+     */
+    private fun tryAbortingUserThreads(threadId: Int, blockingReason: BlockingReason?) {
+        val userThreadsAbortionPossible =
+            // all `TestThread`s are finished (including main: with id of zero)
+            (0 ..< scenario.nThreads).all(threadScheduler::isFinished) &&
+            // The main thread finished its execution (actually all `TestThread`s did): successfully or not, we don't care.
+            // If all user threads (those that are not `TestThread` instances) are not blocked, then abort
+            // the running user threads. Essentially treating them as "daemons", which completion we do not wait for.
+            // Thus, abort all of them and allow `runner` to process the invocation result accordingly.
+            getRegisteredThreads()
+                .map { it.threadId }
+                .filterNot { isTestThread(it) || it == threadId /* we check invoking thread separately */ }
```

Comment:
Could you please extract this part (perhaps even in a method inside `ThreadScheduler`):

```kotlin
val userThreads = ...
```

Reply:
ThreadScheduler does not differentiate between user and test threads. I added a method to the stategy